### ISSUE TITLE
Reloading JS source files included in :require

### ISSUE THREAD

**Issue by alex-dixon** [thheller/shadow-cljs#240]
First time using shadow-cljs. Amazing work.

I'm trying to create a babel example based on the guide in the user manual that explains how to do that using babel directly (until shadow-cljs provides this functionality): https://shadow-cljs.github.io/docs/UsersGuide.html#_javascript_dialects

I've copied the instructions there as best I could: https://github.com/alex-dixon/shadow-cljs-examples/commit/925ced632cc870bfc6023bed53b82a2a0c00b0b7

Unfortunately changing the JS file doesn't reload it and result in an error:
```
Compilation failed!
no source by provide: module$demo$bar
{:provide module$demo$bar}
ExceptionInfo: no source by provide: module$demo$bar
	clojure.core/ex-info (core.clj:4739)
	clojure.core/ex-info (core.clj:4739)
	shadow.build.data/get-source-id-by-provide (data.clj:158)
	shadow.build.data/get-source-id-by-provide (data.clj:155)
	shadow.build.data/get-source-by-provide (data.clj:161)
	shadow.build.data/get-source-by-provide (data.clj:160)
	shadow.build.resolve/resolve-string-require (resolve.clj:222)
	shadow.build.resolve/resolve-string-require (resolve.clj:212)
```
This happens as soon as I change either the source JS file that babel compiles, or the compiled source itself.

Refreshing the browser page fixes the error, but the markup generated by the JS code is stale -- if I change "Woah JSX" to "Woah JSX!" I still see "Woah JSX" on refresh.

The error isn't thrown when modifying `shadow-cljs-examples/babel/public/js/compiled/cljs-runtime/module$demo$bar.js` directly, but a page refresh is required to show the updated markup it generates.

**Comment by thheller** [comment]
The problem is you babel-presets I think. Closure currently does not understand the `__esModule` idiom that babel uses when rewriting files. It thinks the file is CommonJS and rewrites everything to `.default`.

Try only transforming the JSX. The file must stay ES6 for Closure to work currently. Don't know what to do about their CommonJS processing yet. See https://github.com/thheller/shadow-cljs/issues/204
```
{
    "plugins": [
        "transform-react-jsx"
    ]
}
```

**Comment by alex-dixon** [comment]
Thanks. Using this plugin it appears to get further but React is not defined. I made a few changes that may have impacted this. One was removing `shadow.markup.react` and `shadow.markup.dom` from any require statements, upgrading shadow-cljs to ^2.2.25, and react and react-dom to 16.3.1.

I also exported the JS React component using "export default" and imported in CLJS using "default", and simplified the initial rendering by calling the component as a function:
```
(defn root []
  (bar))

(defn ^:export mount-root []
  (render (root) (.getElementById js/document "app")))
```

It now seems React is not defined in the JS context. In the browser I'm getting an error that points to bar.js:4 saying React.createElement is not a function. Seems like source maps are working!:

```
import React from "/node_modules/react/index.js";

function MyComponent() {
  return React.createElement(
    "h1",
    null,
    "Woah JSX!"
  );
}

export default MyComponent;
```

Logging the component from the CLJS file, I get
```
function MyComponent$$module$demo$bar() {
  return module$node_modules$react$index.default.createElement("h1", null, "Woah JSX!");
}
```

This is all behavior when the page initially loads. If I change the JS file, I get the same error as before.

**Comment by alex-dixon** [comment]
Reloading the JS file fixed in 2.2.26.

Changed example to use babel transform-react-jsx. Not sure if this is necessary so will try with babel-preset-env again when I have a chance.

Updated example here: https://github.com/alex-dixon/shadow-cljs-examples/commit/5ffd6dfaafca79cd959e989233b2420e3349fcd4

**Comment by alex-dixon** [comment]
Fixed 95e624702631c11feb1e9b9cd2fda0634f2fd6ee
